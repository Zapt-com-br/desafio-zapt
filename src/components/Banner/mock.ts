export default {
  title: 'Bem vinda Polyane,',
  contactNumber: '11910898780'
}
