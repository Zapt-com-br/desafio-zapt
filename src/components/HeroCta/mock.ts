const heroTitle = 'Bem vindo Marcio,'
const heroSubtitle =
  'Para você aproveitar nossas oportunidades primeiro você precisa estar dentro de um grupo, estamos aqui para lhe ajudar.'
const ctaTitle = 'O que você prefere?'
const firstOptionLabel = 'Entrar em um grupo'
const secondOptionLabel = 'Criar um grupo'

export default {
  heroTitle,
  ctaTitle,
  heroSubtitle,
  firstOptionLabel,
  secondOptionLabel
}
