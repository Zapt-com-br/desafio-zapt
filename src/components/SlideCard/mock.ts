const userComment = {
  content:
    '”Não consigo mais viver sem os produtos do Sítio, a diferença do que eu comprava antes e agora é abissal”',
  user: {
    name: 'Patricia Godoy',
    location: 'São Paulo • São Paulo',
    icon: '/img/users-comments/woman-icon.png'
  },
  backgroundImage: '/img/users-comments/woman-image.png'
}

export default userComment
