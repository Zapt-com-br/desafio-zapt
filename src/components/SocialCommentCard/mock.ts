export default {
  content:
    '”Não da para parar de comer.. esse final de semana pedi 20 pasteis da Ilma e acabaram TODOS em um só dia... O pessoal aqui de casa ficou viciado de um nível que terei que comprar toda semana... Assim não tem academia que secura rsrsrsrs Obrigado Zapt.”',
  user: {
    profileIcon: '/img/users-icon/user-icon1.png',
    username: '@ferri-gui'
  }
}
