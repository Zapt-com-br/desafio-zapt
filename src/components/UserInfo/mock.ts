import { UserInfoProps } from '.';

export const userInfo: UserInfoProps = {
  avatarUrl: 'https://i.imgur.com/8RP8XA9.png',
  name: 'Patricia Godoy',
  city: 'São Paulo',
  state: 'São Paulo',
};
