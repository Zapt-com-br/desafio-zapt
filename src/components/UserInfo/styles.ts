import styled, { css } from 'styled-components';
import media from 'styled-media-query';

export const Wrapper = styled.div`
  display: flex;
  align-items: center;
`;

export const Avatar = styled.img`
  ${({ theme }) => css`
    width: 3rem;
    height: 3rem;
    border-radius: 50%;
    border: 0.15rem solid ${theme.colors.white};

    ${media.greaterThan('small')`
      width: 4rem;
      height: 4rem;
    `}
  `}
`;

export const Info = styled.div`
  ${({ theme }) => css`
    color: ${theme.colors.white};
    margin-left: 1.6rem;
  `}
`;

export const Name = styled.h3`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    font-weight: ${theme.font.normal};

    ${media.greaterThan('small')`
      font-size: ${theme.font.sizes.medium};
    `}
  `}
`;

export const City = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xxsmall};
    font-weight: ${theme.font.normal};

    ${media.greaterThan('small')`
      font-size: ${theme.font.sizes.xsmall};
    `}
  `}
`;
