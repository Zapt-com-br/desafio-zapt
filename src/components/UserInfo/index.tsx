import * as S from './styles';

export type UserInfoProps = {
  avatarUrl: string;
  name: string;
  city: string;
  state: string;
};

const UserInfo = ({ avatarUrl, name, city, state }: UserInfoProps) => (
  <S.Wrapper>
    <S.Avatar src={avatarUrl} alt={name} />

    <S.Info>
      <S.Name>{name}</S.Name>
      <S.City>
        {city} â€¢ {state}
      </S.City>
    </S.Info>
  </S.Wrapper>
);

export default UserInfo;
