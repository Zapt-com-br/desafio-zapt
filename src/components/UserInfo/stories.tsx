import { Story, Meta } from '@storybook/react/types-6-0';

import UserInfo, { UserInfoProps } from '.';
import { userInfo } from './mock';

export default {
  title: 'UserInfo',
  component: UserInfo,
  parameters: {
    backgrounds: {
      default: 'dark',
    },
  },
} as Meta;

export const Default: Story<UserInfoProps> = (args) => <UserInfo {...args} />;

Default.args = { ...userInfo };
