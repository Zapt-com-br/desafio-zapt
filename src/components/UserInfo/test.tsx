import { render, screen } from 'utils/test-utils';

import UserInfo from '.';

import { userInfo } from './mock';

describe('<UserInfo />', () => {
  it('should match snapshot', () => {
    const { container } = render(<UserInfo {...userInfo} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with avatar image', () => {
    const avatarUrl = 'some-url';
    const name = 'user name';

    render(<UserInfo {...userInfo} avatarUrl={avatarUrl} name={name} />);

    expect(screen.getByRole('img', { name })).toHaveAttribute('src', avatarUrl);
  });

  it('should render with user name', () => {
    const name = 'user name';

    render(<UserInfo {...userInfo} name={name} />);

    expect(screen.getByRole('heading', { name })).toBeInTheDocument();
  });

  it('should render with city and state', () => {
    const city = 'Belo Horizonte';
    const state = 'Minas Gerais';

    render(<UserInfo {...userInfo} city={city} state={state} />);

    expect(screen.getByText(`${city} â€¢ ${state}`)).toBeInTheDocument();
  });
});
