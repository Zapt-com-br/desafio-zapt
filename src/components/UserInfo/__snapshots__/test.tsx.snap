// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserInfo /> should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  width: 3rem;
  height: 3rem;
  border-radius: 50%;
  border: 0.15rem solid #ffffff;
}

.c2 {
  color: #ffffff;
  margin-left: 1.6rem;
}

.c3 {
  font-size: 1.4rem;
  font-weight: 400;
}

.c4 {
  font-size: 1.0rem;
  font-weight: 400;
}

@media (min-width:450px) {
  .c1 {
    width: 4rem;
    height: 4rem;
  }
}

@media (min-width:450px) {
  .c3 {
    font-size: 1.6rem;
  }
}

@media (min-width:450px) {
  .c4 {
    font-size: 1.2rem;
  }
}

<div
  class="c0"
>
  <img
    alt="Patricia Godoy"
    class="c1"
    src="https://i.imgur.com/8RP8XA9.png"
  />
  <div
    class="c2"
  >
    <h3
      class="c3"
    >
      Patricia Godoy
    </h3>
    <p
      class="c4"
    >
      São Paulo
       • 
      São Paulo
    </p>
  </div>
</div>
`;
